### ISSUE TITLE
Required puppet version ?

### ISSUE THREAD

**Issue by Kyoshiro-san** [lvicainne/puppet-opendkim#10]
Hi,

Which puppet version is required for the module to work ?

I tried it before #5 and it complained about that line:
> Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Syntax error at '.'; expected '}' at /etc/puppet/modules/opendkim/manifests/config.pp:114 

Which was containing:
> $opendkim::keys.each

Now that #5 has been merged, it fails with:
> Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Syntax error at 'String'; expected ')' at /etc/puppet/modules/opendkim/manifests/init.pp:7

Is it compatible with debian jessie (oldstable) version 3.7.2-4 ?
As stretch is fairly recent, it might not be compatible with most debian user's puppet installs.

Maybe you could tag different revisions of this module with some milestones / version compatibility ?

Anyways, thanks for your work.
Cheers

Loïc

**Comment by lvicainne** [comment]
Hello @Kyoshiro-san 
Thanks for your attention !
You are fully true : the manifest was not up-to-date compare to the last pull-request (I mean, we are no more compatible with Puppet older than 4.8).
Thus, i built today a new branch : 
* branch puppet-v4.0 for older puppet version (<= 4.6)
* branch master for newer versions of puppet. (> 4.8 and newer).
To follow this new evolution, I created also a new nomenclature for the versions of this module : 
v0.1.* are compatible with Puppet versions up to 4.6 (at least for v0.1.2)
v0.2.* are comaptible with newer versions of Puppet (v4.8, 5.0 at least)

I expect you will find your happiness :-)
Feel free to comment ;-) I took some time between your issue and its solve 
Rgds
